File: c/tree/maximum-depth-of-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
void depth(struct TreeNode *root, int cur_h, int *max_h)
{
    if(root == NULL)
        return ;

    cur_h++;
    if(cur_h > *max_h)
        *max_h = cur_h;

    depth(root->left, cur_h, max_h);
    depth(root->right, cur_h, max_h);
}
int maxDepth(struct TreeNode* root) {
    int max_h = 0;
    int cur_h = 0;

    depth(root, cur_h, &max_h);

    return max_h;
}

File: python/tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def depth(self,root,cur_h):
        if root is None:
            return

        cur_h += 1
        if cur_h > self.max_h:
            self.max_h = cur_h

        self.depth(root.left,cur_h)
        self.depth(root.right,cur_h)

    def maxDepth(self, root):
        self.max_h = 0
        self.depth(root,0)

        return self.max_h
